File: 프로그래머스/2/42578. 의상/의상.js
function solution(clothes) {
  var answer = 1;
  const clothesMap = new Map();
  for (const [clothName, clothCategory] of clothes) {
    clothesMap.set(clothCategory, (clothesMap.get(clothCategory) || 0) + 1); // 누적으로 더함. 최종적으로 +1 을 한 것은, 아무것도 입지 않는 경우도 고려해야 하기 때문
  }
  for (let clo of clothesMap) {
    answer *= clo[1] + 1;
  }
  return answer - 1;
}

File: 프로그래머스/2/42578. 의상/의상.py
# 여기의 핵심은 수학 계산 식
# 의상종류 + 1 => 여기서 1은 안입는 경우이다. 마지막에 뺄셈하는 -1은 모두 벗은경우
# (a의상종류 + 1) * (b의상종류 +1) * ... - 1

def solution(clothes):
    dic = {}
    res = 1
    for i in clothes:
        key = i[1]
        dic[key] = dic.get(key,1) + 1
    for j in dic:
        res *= dic[j]
    return res - 1
